amazon/ion/writer_text.py

```diff
@@ -173,15 +173,22 @@ def _bytes_datetime(dt):
         return tz_string + _bytes_utc_offset(dt)
 
     fractional_precision = getattr(original_dt, TIMESTAMP_FRACTION_PRECISION_FIELD, MICROSECOND_PRECISION)
-    if fractional_precision:
+    fractional_seconds = getattr(original_dt, TIMESTAMP_FRACTIONAL_SECONDS_FIELD, None)
+    if fractional_seconds is None:
         fractional = dt.strftime('%f')
         assert len(fractional) == MICROSECOND_PRECISION
-        if fractional[fractional_precision:] != ('0' * (MICROSECOND_PRECISION - fractional_precision)):
-            raise ValueError('Found timestamp fractional with more than the specified %d digits of precision.'
-                             % (fractional_precision,))
-        fractional = fractional[:fractional_precision]
-        tz_string += '.' + fractional
 
+        if fractional_precision is not None:
+            if fractional[fractional_precision:] != ('0' * (MICROSECOND_PRECISION - fractional_precision)):
+                raise ValueError('Found timestamp fractional with more than the specified %d digits of precision.'
+                                 % (fractional_precision,))
+            fractional = fractional[:fractional_precision]
+            tz_string += '.' + fractional
+
+    if fractional_seconds == 0:
+        tz_string += '.' + ('0' * fractional_precision)
+    elif fractional_seconds is not None:
+        tz_string += str(fractional_seconds)[1:]
```

Comment:
Is `str(fractional_seconds)` always guaranteed to give a string representation that doesn't contain 'e'? Even for a decimal value with precision of, say, 500?

Reply:
Good catch. Added a check for this, and some logic to handle this scenario contained within a new method called `scientific_notation_to_decimal_string()`